x-pack/plugins/alerting/server/invalidate_pending_api_keys/task.ts

```diff
@@ -189,37 +157,172 @@ function taskRunner(
   };
 }
 
-async function invalidateApiKeys(
-  logger: Logger,
-  savedObjectsClient: SavedObjectsClientContract,
-  apiKeysToInvalidate: SavedObjectsFindResponse<InvalidatePendingApiKey>,
-  encryptedSavedObjectsClient: EncryptedSavedObjectsClient,
-  securityPluginStart?: SecurityPluginStart
-) {
+interface ApiKeyIdAndSOId {
+  id: string;
+  apiKeyId: string;
+}
+
+interface RunInvalidateOpts {
+  config: AlertingConfig;
+  encryptedSavedObjectsClient: EncryptedSavedObjectsClient;
+  logger: Logger;
+  savedObjectsClient: SavedObjectsClientContract;
+  security?: SecurityPluginStart;
+}
+export async function runInvalidate({
+  config,
+  encryptedSavedObjectsClient,
+  logger,
+  savedObjectsClient,
+  security,
+}: RunInvalidateOpts) {
+  const configuredDelay = config.invalidateApiKeysTask.removalDelay;
+  const delay: string = timePeriodBeforeDate(new Date(), configuredDelay).toISOString();
+
+  let hasApiKeysPendingInvalidation = true;
   let totalInvalidated = 0;
+  const excludedSOIds = new Set<string>();
+  do {
+    const filter = getFindFilter(delay, [...excludedSOIds]);
+    const apiKeysToInvalidate = await savedObjectsClient.find<InvalidatePendingApiKey>({
+      type: API_KEY_PENDING_INVALIDATION_TYPE,
+      filter,
+      page: 1,
+      sortField: 'createdAt',
+      sortOrder: 'asc',
+      perPage: PAGE_SIZE,
+    });
+
+    if (apiKeysToInvalidate.total > 0) {
+      const { apiKeyIdsToExclude, apiKeyIdsToInvalidate } = await getApiKeyIdsToInvalidate({
+        apiKeySOsPendingInvalidation: apiKeysToInvalidate,
+        encryptedSavedObjectsClient,
+        savedObjectsClient,
+      });
+      apiKeyIdsToExclude.forEach(({ id }) => excludedSOIds.add(id));
+      totalInvalidated += await invalidateApiKeysAndDeletePendingApiKeySavedObject({
+        apiKeyIdsToInvalidate,
+        logger,
+        savedObjectsClient,
+        securityPluginStart: security,
+      });
+    }
+
+    hasApiKeysPendingInvalidation = apiKeysToInvalidate.total > PAGE_SIZE;
```

Comment:
We need the variable because `apiKeysToInvalidate` doesn't exist outside of the `do...while` loop so I updated the variable name in https://github.com/elastic/kibana/pull/180749/commits/cf85e78edcb813d46da2c53922ad9478df597111